packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -2454,7 +2454,12 @@ export class StripeHelper {
    */
   async processCustomerEventToFirestore(event: Stripe.Event) {
     const { data } = event;
-    const customer = data.object as Stripe.Customer;
+    const customer = await this.stripe.customers.retrieve(
+      (data.object as Stripe.Customer).id
+    );
+    if (customer.deleted) {
```

Comment:
hmmm I noticed that the other set of handlers does not handle `customer.deleted`, so if we just return early here, we won't ever know that a customer has been deleted I think.

Reply:
Ah, good catch, because we will not have touched the record in Firestore.